server/functions/framework/common_type.go

```diff
@@ -37,16 +39,16 @@ func FindCommonType(types []pgtypes.DoltgresTypeBaseID) (pgtypes.DoltgresTypeBas
 	}
 	if !fail {
 		if candidateType == pgtypes.DoltgresTypeBaseID_Unknown {
-			return pgtypes.DoltgresTypeBaseID_Text, true
+			return pgtypes.DoltgresTypeBaseID_Text, nil
 		}
-		return candidateType, true
+		return candidateType, nil
 	}
 	for _, typBaseID := range types {
 		if candidateType == pgtypes.DoltgresTypeBaseID_Unknown {
 			candidateType = typBaseID
 		}
 		if typBaseID != pgtypes.DoltgresTypeBaseID_Unknown && candidateType.GetTypeCategory() != typBaseID.GetTypeCategory() {
-			return 0, false
+			return 0, fmt.Errorf("ARRAY types %s and %s cannot be matched", candidateType.GetRepresentativeType().String(), typBaseID.GetRepresentativeType().String())
```

Comment:
This will be called from other expressions besides the `ARRAY` expression, where this error message wouldn't make as much sense. You can return some kind of error information that's not an error message, to allow the expression to craft their own error messages.